src/SAWScript/Crucible/LLVM/ResolveSetupValue.hs

```diff
@@ -86,17 +86,24 @@ resolveSetupValueInfo ::
   L.Info                          {- ^ field index       -}
 resolveSetupValueInfo cc env nameEnv v =
   case v of
-    -- SetupGlobal g ->
+    SetupGlobal _ name ->
+      case lookup (L.Symbol name) globalTys of
+        Just (L.Alias alias) -> L.Pointer (guessAliasInfo mdMap alias)
+        _ -> L.Unknown
+
     SetupVar i
       | Just alias <- Map.lookup i nameEnv
-      , let mdMap = Crucible.llvmMetadataMap (ccTypeCtx cc)
       -> L.Pointer (guessAliasInfo mdMap alias)
+
     SetupField () a n ->
        fromMaybe L.Unknown $
        do L.Pointer (L.Structure xs) <- return (resolveSetupValueInfo cc env nameEnv a)
           listToMaybe [L.Pointer i | (n',_,i) <- xs, n == n' ]
 
     _ -> L.Unknown
+  where
+    globalTys = [ (L.globalSym g, L.globalType g) | g <- L.modGlobals (ccLLVMModuleAST cc) ]
```

Comment:
it would be nice if this was a map

Reply:
It's probably only worth building the map if we save it somewhere to reuse in later calls. Otherwise it's probably more efficient to do a single linear scan of the list, rather than build a map and then do a single lookup before throwing the map away.

Reply:
that's what I meant, it would be nice if `modGlobals` was a map. As it is, I agree that a linear scan is faster